### PR TITLE
Vehicles CRUD Functionality

### PR DESCRIPTION
**Pull request by hsmitha26**
This PR does the following:
- Adds all CRUD functionality for vehicles that belong to a user who is logged in.
- Adds Vehicle model and tests.
- Updates datatbase schema with vehicles table.
- Features working in development.
- All tests passing.

### CODE REVIEW

File: app/controllers/vehicles_controller.rb [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+      redirect_to user_vehicles_path(current_user.id), alert: 'Something went wrong. Please try again.'
+    end
+  end
+private
+
+  def vehicle_params
+    params.require(:vehicle).permit(:make, :model, :year, :awd, :storage_rack, :total_seats, :current_user)
+  end
+
+end
```
Comment (leiyakenney) [inline app/controllers/vehicles_controller.rb:54]: Nice! I think we can probably refactor a bit of this to make our controller even skinnier if we have time, but looks good!

File: app/models/vehicle.rb [hunk 2]
```diff
@@ -0,0 +1,6 @@
+class Vehicle < ApplicationRecord
+  belongs_to :user
+
+#I do not think we want to validate presence of storage_rack and awd. I think this would mean that if a user does not have one of these they will be unable to add their car. Maybe? Something to test for.
+  validates_presence_of :make, :model, :year, :total_seats
+end
```
Comment (leiyakenney) [inline app/models/vehicle.rb:6]: Agreed on the comment - I don't think we need to validate for that; they're more optional fields anyways.

File: app/views/layouts/application.html.erb [hunk 3]
```diff
@@ -37,33 +37,28 @@
       <li><strong><i class="fas fa-car-side"></i><%= link_to "My Rides", "#"%></strong></li>
 
         <% if user_signed_in? %>
-        <li><strong><i class="fas fa-sign-out-alt"></i>
+          <li><strong><i class="fas fa-user"></i>
```
Comment (leiyakenney) [inline app/views/layouts/application.html.erb:40]: I'm sure this has something to do with Devise, but can you explain the "fas fa-xx" stuff to me? Also the <strong> tag? 

i.e. when I'm creating my reviews classes should I also have that stuff in my class names? I was just naming them like we did for Monster Shop with something like "review-#{review_id}"

File: app/views/users/show.html.erb [hunk 4]
```diff
@@ -32,7 +38,7 @@
   <%= @user.bio %>
 </p>
 
-<%= link_to 'Edit', edit_user_path(@user) %> |
-<%= link_to 'Back', users_path %>
+<%= link_to 'Edit', edit_user_path(@user.id) %> | 
+<%= link_to 'Add A Vehicle', new_user_vehicle_path(@user.id) %>
 
 </div>
```
Comment (leiyakenney) [inline app/views/users/show.html.erb:44]: What would you think about adding links to the choose ride page from the profile? I don't know if it's a good idea or not - just trying to brainstorm some user experience stuff.

File: app/views/vehicles/edit.html.erb [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+  </div>
+
+  <div class="field">
+    <%= form.label :storage_rack %>
+    <%= form.check_box :storage_rack %>
+  </div>
+
+  <div class="field">
+    <%= form.label :total_seats %>
+    <%= form.number_field :total_seats, min: 1, required: true %>
```
Comment (leiyakenney) [inline app/views/vehicles/edit.html.erb:30]: Ooh I like this min: 1 requirement! Nice thinking!

File: app/views/vehicles/index.html.erb [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+<% @vehicles.each do |vehicle|%>
+  <ul class='vehicle'>
+    <div class='car-icon'>
+      <% if vehicle.image == nil %>
+        <%= link_to image_tag("https://www.chalets-meribel.co.uk/images/graphics/car-journey.png", width: 50, height: 50), user_vehicle_path(current_user.id, vehicle.id) %>
+      <% else %>
+        <img> <%= image_tag(vehicle.image, width: 50, height: 50)%>
+        </img>
+      <% end %>
+    </div>
```
Comment (leiyakenney) [inline app/views/vehicles/index.html.erb:15]: Nice - I like this

File: app/views/vehicles/index.html.erb [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+      <div class="storage-rack">
+        <p>Storage Rack: <i class="far fa-check-square"></i></p>
+      </div>
+    <% else %>
+      <div class="storage-rack">
+        <p>Storage Rack: <i class="far fa-square"></i></p>
+      </div>
+    <% end %>
+    <p class='total-seats'>Total Seats: <%=vehicle.total_seats%></p>
+  </ul>
```
Comment (leiyakenney) [inline app/views/vehicles/index.html.erb:39]: Should we specify whether this is including the driver or not?

File: app/views/vehicles/new.html.erb [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+
+  <div class="field">
+    <%= form.label :storage_rack %>
+    <%= form.check_box :storage_rack %>
+  </div>
+
+  <div class="field">
+    <%= form.label :total_seats %>
+    <%= form.number_field :total_seats, min: 1, required: true %>
+  </div>
```
Comment (leiyakenney) [inline app/views/vehicles/new.html.erb:31]: Repeating myself, but I just thought that we might want to be super clear when having drivers fill this out that this includes the driver (I think...?)

File: config/routes.rb [hunk 9]
```diff
@@ -2,7 +2,9 @@
   # devise_for :users
   devise_for :users, controllers: { sessions: 'users/sessions', registrations: 'users/registrations' }
 
-  resources :users
+  resources :users do
+    resources :vehicles
+  end
```
Comment (leiyakenney) [inline config/routes.rb:7]: Not a comment about your code at all, but I'm going to have a merge conflict here cause I have a reviews section, but that's not a big deal. I also moved root to: up to the top underneath the devise_for, cause I feel like that's more logical organization. Again, not a big deal, more of an FYI

File: spec/features/a_user_can_log_in_and_out_spec.rb [hunk 10]
```diff
@@ -24,7 +24,7 @@
   end
 
   it "I can log out and sign back in" do
-    within('.nav-bar') do
+    within('#menuToggle') do
       click_on 'Logout'
```
Comment (leiyakenney) [inline spec/features/a_user_can_log_in_and_out_spec.rb:28]: I haven't seen this before - what does it do?

File: spec/models/user_spec.rb [hunk 11]
```diff
@@ -8,4 +8,8 @@
     it {should validate_presence_of(:last_name)}
     it {should validate_presence_of(:email)}
   end
+
+  describe "Relationships" do
+    it {should have_many :vehicles}
+  end
```
Comment (leiyakenney) [inline spec/models/user_spec.rb:14]: Again, I'll have a minor merge conflict here, but not a big deal.

**Review by leiyakenney — APPROVED** [review]
Looks good to me - I added a couple questions/comments to the files that were changed. Also, this passed the build for Travis CI, so that's a good sign! Nice testing.

File: app/views/vehicles/index.html.erb (hunk 7 above)
Comment (hsmitha26) [reply app/views/vehicles/index.html.erb:39]: Yes, we should.  Sorry, I forgot about that.

File: app/controllers/vehicles_controller.rb (hunk 1 above)
Comment (nathangthomas) [reply app/controllers/vehicles_controller.rb:54]: Key words "if we have time." I totally agree though!

File: app/views/users/show.html.erb (hunk 4 above)
Comment (hsmitha26) [reply app/views/users/show.html.erb:44]: All agreed that 'Book A Ride' will go on profile page and hamburger menu

File: app/views/layouts/application.html.erb (hunk 3 above)
Comment (nathangthomas) [reply app/views/layouts/application.html.erb:40]: user_signed_in? is a devise method. <i class="fas fa-sign-out-alt"></i> is FontAwesome code that we used for icons. 
You can name your review pages however you like depending on how you are would like to style the page or how you are writing your feature tests. I'm also fine with waiting and doing styling together when we can work together in person.

File: app/views/users/show.html.erb (hunk 4 above)
Comment (nathangthomas) [reply app/views/users/show.html.erb:44]: Good thinking! I think we should probably have a link to set up a ride both on the profile page as well as in the hamburger menu considering that is our apps primary purpose.

File: spec/features/a_user_can_log_in_and_out_spec.rb (hunk 10 above)
Comment (nathangthomas) [reply spec/features/a_user_can_log_in_and_out_spec.rb:28]: #menuToggle doesn't do anything special is just defines of our css ids for styling of the hamburger menu.

**Pull request merged**